### ISSUE TITLE
It's widgets are ativated when collapsed window clicked or moved.

### ISSUE THREAD

**Issue by demucin** [ocornut/imgui#4603]
<p><strong>Version/Branch of Dear ImGui:</strong><br>
Version: v1.85<br>
Branch: master</p>
<p><strong>Back-end/Renderer/Compiler/OS:</strong><br>
Back-ends: imgui_impl_opengl3.cpp + imgui_impl_glfw.cpp<br>
gcc/windows</p>
<p><strong>My Issue/Question:</strong><br>
When collapsed(minimized) window's title clicked<br>
widgets in the window are activated.<br>
As I can see <code>if(ImGui::Begin("..."))</code><br>
Condition (if) solves this issue.<br>

I'm sory if it's explained in somewhere in source code or issues.<br>
I wonder why ImGui needs to process collapsed window's content.

<p><strong>minimal code to reproduce:</strong><br>
Collapse the window end try to move it.<br>
<br>

</p>
<code>
        ImGui::Begin("widgets-activate test");
            ImGui::Button("button", ImVec2(64, 24));
            if(ImGui::IsItemActive())
                ImGui::GetForegroundDrawList()->AddCircleFilled(
                    ImVec2(ImGui::GetMousePos().x, ImGui::GetMousePos().y), 32,
                    ImGui::GetColorU32(IM_COL32(255, 0, 255, 155)));

            static float sf = .0f;
            ImGui::SliderFloat("sliderfloat", &sf, 0.0f, 1.0f);
            if(ImGui::IsItemActive())
                ImGui::GetForegroundDrawList()->AddCircleFilled(
                    ImVec2(ImGui::GetMousePos().x, ImGui::GetMousePos().y), 24,
                    ImGui::GetColorU32(IM_COL32(255, 255, 0, 155)));

            static bool cb = true;
            ImGui::Checkbox("checkbox", &cb);
            if(ImGui::IsItemActive())
                ImGui::GetForegroundDrawList()->AddCircleFilled(
                    ImVec2(ImGui::GetMousePos().x, ImGui::GetMousePos().y), 16,
                    ImGui::GetColorU32(IM_COL32(0, 255, 0, 155)));
        ImGui::End();
</code>

**Comment by ocornut** [comment]
That's correct, there's an issue as marking the window contents as fully clipped (`Begin()` returned false here) widgets are earlying out.

Note that you should be testing Begin()'s return value to avoid submitting contents.
But I think we should still find a way to fix this issue.

**Comment by ocornut** [comment]
The following functions are affected
- IsItemActive
- IsItemActivated
- IsItemDeactivated
- SetItemAllowOverlap
- SetItemUsingMouseWheel

The following functions already have a custom workaround
- IsItemHovered
- IsItemFocused (docking branch only)

```cpp
// Special handling for calling after Begin() which represent the title bar or tab.
// When the window is collapsed (SkipItems==true) that last item will never be overwritten so we need to detect the case.
if (g.LastItemData.ID == window->MoveId && window->WriteAccessed)
    return false;
```

I think the proper fix should be that the function setting WriteAccessed=true may want to clear LastItemID but it's a bit awkward because effectively with existing widget structure that function is called `GetCurrentWindow()` and the widget structure is likely copy and pasted into user's custom widget. I'll find a suitable solution.

**Comment by ocornut** [comment]
There are other subtle ramifications to this issue that may be trickier to fix.

I am thinking maybe this is something that could wait until the new Begin/End api for which it'll become a non-issue.
(Basically we have near-ready work to change Begin/BeginChild to REQUIRES user testing for the bool, it would be a major API break but fix a lots of inconsistency and simplify some problem. See proto at https://github.com/ocornut/imgui/commits/features/new_begin_end)

From your POV today if you test the return value of `Begin()` you won't have any issue.

**Comment by demucin** [comment]
Thank you for the response.
I just wanted to inform you.

I already use your style in "ImGui_Demo.cpp":
<code>
    if (!ImGui::Begin(...))
    {
        ImGui::End();
        return;
    }
</code>
